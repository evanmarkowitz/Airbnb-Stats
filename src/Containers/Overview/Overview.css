.overview {
  margin-right: 15px;
  min-width: 250px;
  border-bottom: 1px lightgrey solid;
  border-top: 1px lightgrey solid;
}