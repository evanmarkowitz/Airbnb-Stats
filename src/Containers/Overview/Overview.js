import React from 'react'
import { connect } from 'react-redux'
import './Overview.css'


export const Overview = ({hood, aptType, apts}) => {
  let currentAptType = aptType === '' ? 'All' : aptType
  let currentHood = hood === '' ? 'All' : hood

  const filterApartments = () => {
    let aptsByHood = apts
    if (hood !== ''){
      aptsByHood = apts.filter(apt => apt.hood === hood)
    } 
    let aptsByRoomType = aptsByHood
    if (aptType !== '') {
      aptsByRoomType = aptsByRoomType.filter(apt => apt.aptType === aptType)
    }
    return aptsByRoomType
  }

  const calcAvgPrice = () => {
    let filteredApts = filterApartments()
    let total = filteredApts.reduce((total, apt) => {
      total = total + apt.price 
      return total 
    },0)
    let avg = Math.floor(total / filteredApts.length)
    if (isNaN(avg)) {
      avg = 'Not enough data'
    }
    return avg
  }

  return (
    <section className='overview'>
      <h3>Filtered By:</h3>
      <p><span className='key'>Apartment Type:</span> {currentAptType}</p>
      <p><span className='key'>Neighborhood:</span> {currentHood}</p>
      <p><span className='key'>Average Price:</span> {calcAvgPrice()}</p>
    </section>
  )
}

export const mapStateToProps = (state) => ({
  hood: state.hood,
  aptType: state.aptType,
  apts: state.apts
})

export default connect(mapStateToProps)(Overview)



