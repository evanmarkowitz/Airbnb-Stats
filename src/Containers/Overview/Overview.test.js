import {mapStateToProps, Overview} from './Overview'
import {shallow} from 'enzyme'
import React from 'react'


describe('Overview', () => {
  let apts;
  let aptType;
  let hood;
  let wrapper;

  beforeEach(() => {  
    hood = 'Williamsburg'
    aptType = 'Private Room'
    apts = [{reviewsPerMonth: 1.5, id: 8252369, numReviews: 68,
      price: 85, aptType: "Private room", name: "APT 2",
      hood: "Williamsburg", lat: 40, long: -73},
      {reviewsPerMonth: 1.5, id: 8252369, numReviews: 68,
        price: 85, aptType: "Entire home/apt", name: "APT 1",
        hood: "Williamsburg", lat: 40, long: -73},
      {reviewsPerMonth: 1.5, id: 8252369, numReviews: 68,
        price: 85, aptType: "Entire home/apt", name: "APT 1",
        hood: "Greenpoint", lat: 40, long: -73}]
    wrapper = shallow(<Overview  hood={hood} aptType={aptType} apts={apts} />)
  })

  
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map state to props', () => {
    let initialState = {hood: hood, aptType: aptType, apts: apts}
    const mappedProps = mapStateToProps(initialState);
    expect(mappedProps).toEqual(initialState);
  })
 
})