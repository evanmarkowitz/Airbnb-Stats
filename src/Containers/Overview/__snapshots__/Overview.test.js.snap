// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should match a snapshot 1`] = `ShallowWrapper {}`;
